src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperDescriptorProvider.cs

```diff
@@ -95,39 +96,107 @@ public TagHelperDescriptorProvider(IEnumerable<TagHelperDescriptor> descriptors)
+                        // At least one attribute where a name matches, now match the value (if restricted).
+                        if (isSelector)
+                        {
+                            var equalsIndex = requiredAttribute.IndexOf('=');
+                            var afterEquals = requiredAttribute[equalsIndex + 1];
+                            var valueOffset = 0;
+                            if (afterEquals == '\'' || afterEquals == '"')
+                            {
+                                valueOffset = 1;
+                            }
+                            var valueStart = equalsIndex + 1 + valueOffset;
+                            var valueLength = requiredAttribute.Length - equalsIndex - valueOffset * 2 - 2;
+                            var requiredValue = requiredAttribute.Substring(valueStart, valueLength);
+                            var selectorCharacter = requiredAttribute[equalsIndex - 1];
+
+                            switch (selectorCharacter)
+                            {
+                                case '^': // Value starts with
+                                    if (!matchingAttributes.Any(attribute => attribute.Value.StartsWith(requiredValue, StringComparison.Ordinal)))
+                                    {
+                                        return false;
+                                    }
+                                    break;
+                                case '$': // Value ends with
+                                    if (!matchingAttributes.Any(attribute => attribute.Value.EndsWith(requiredValue, StringComparison.Ordinal)))
+                                    {
+                                        return false;
+                                    }
+                                    break;
+                                default: // Value equals
+                                    if (!matchingAttributes.Any(attribute => string.Equals(attribute.Value, requiredValue, StringComparison.Ordinal)))
+                                    {
+                                        return false;
+                                    }
+                                    break;
+                            }
+                        }
                     }
 
                     return true;
                 });
         }
 
+        public static string GetRequiredAttributeName(string originalAttributeName)
+        {
+            if (originalAttributeName.Length <= 2 || originalAttributeName[0] != '[')
+            {
+                return originalAttributeName;
+            }
+
+            string name;
+
+            // CSS Selector
+            var equalsIndex = originalAttributeName.IndexOf('=');
+            if (equalsIndex == -1)
+            {
+                // Can assume there's a matching ']' at the end because it has run through validation already.
+                name = originalAttributeName.Substring(1, originalAttributeName.Length - 2);
```

Comment:
Too strict. CSS is perfectly fine w/ whitespace before and after the identifier in an attribute selector.